### ISSUE TITLE
noexec seems to be interfering with PlayOnLinux

### ISSUE THREAD

**Issue by jacobpewitt** [dnschneid/crouton#528]
Hello,

Please forgive me, as I am pretty new to Linux and still have a lot to learn.

I have crouton installed on my Acer C720, raring with cinnamon. I am running into an issue while trying to install Warcraft 3 via PlayOnLinux. Every time I try to install it, I get an error saying that the installation file is in a noexec mounted drive which may prevent Wine from running properly. It will go through the installation process, but when I actually try to open the game it instantly crashes. 

After searching through other issues, I came upon this one:

https://github.com/dnschneid/crouton/issues/26

I noticed that I had been installing PlayOnLinux from a .deb file in my downloads folder, and thought maybe that was the problem, so I moved it to my desktop and reinstalled. Same error. The installation files for Warcraft 3 were in folders on my desktop. So I tried moving POL and W3 installation files to / and reinstalling. Same problem. Even when the installation file is in /Warcraft it still says it is located in a noexec mounted drive.

Here is the output of mount from outside the chroot:

```
rootfs on / type rootfs (rw)
/dev/root on / type ext2 (ro,relatime)
devtmpfs on /dev type devtmpfs (rw,nosuid,noexec,relatime,size=1992108k,nr_inodes=498027,mode=755)
none on /proc type proc (rw,nosuid,nodev,noexec,relatime)
none on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
tmp on /tmp type tmpfs (rw,nosuid,nodev,noexec,relatime)
debugfs on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime,gid=236,mode=750)
shmfs on /dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620)
/dev/sda1 on /mnt/stateful_partition type ext4 (rw,nosuid,nodev,noexec,relatime,commit=600,data=ordered)
/dev/sda8 on /usr/share/oem type ext4 (ro,nosuid,nodev,noexec,relatime,commit=600,data=ordered)
/dev/sda1 on /home type ext4 (rw,nosuid,nodev,noexec,relatime,commit=600,data=ordered)
/dev/mapper/encstateful on /mnt/stateful_partition/encrypted type ext4 (rw,nosuid,nodev,noexec,relatime,discard,commit=600,data=ordered)
/dev/mapper/encstateful on /var type ext4 (rw,nosuid,nodev,noexec,relatime,discard,commit=600,data=ordered)
/dev/mapper/encstateful on /home/chronos type ext4 (rw,nosuid,nodev,noexec,relatime,discard,commit=600,data=ordered)
varrun on /var/run type tmpfs (rw,nosuid,nodev,noexec,relatime,mode=755)
varlock on /var/lock type tmpfs (rw,nosuid,nodev,noexec,relatime)
media on /media type tmpfs (rw,nosuid,nodev,noexec,relatime)
/dev/sda1 on /usr/local type ext4 (rw,nodev,relatime,commit=600,data=ordered)
none on /sys/fs/cgroup type tmpfs (rw,nosuid,nodev,noexec,relatime,mode=755)
cgroup on /sys/fs/cgroup/cpu type cgroup (rw,nosuid,nodev,noexec,relatime,cpu)
none on /dev/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
debugfs on /var/run/debugfs_gpu type debugfs (rw,nosuid,nodev,noexec,relatime,gid=236,mode=750)
fusectl on /sys/fs/fuse/connections type fusectl (rw,nosuid,nodev,noexec,relatime)
/home/.shadow/a872706fda597de3825ae97a0ce39c3f6d557f3d/vault on /home/.shadow/a872706fda597de3825ae97a0ce39c3f6d557f3d/mount type ecryptfs (rw,nosuid,nodev,noexec,relatime,ecryptfs_sig=1a860241d16eec64,ecryptfs_fnek_sig=5316ba57b4965925,ecryptfs_cipher=aes,ecryptfs_key_bytes=16,ecryptfs_unlink_sigs)
/home/.shadow/a872706fda597de3825ae97a0ce39c3f6d557f3d/vault on /home/chronos/user type ecryptfs (rw,nosuid,nodev,noexec,relatime,ecryptfs_sig=1a860241d16eec64,ecryptfs_fnek_sig=5316ba57b4965925,ecryptfs_cipher=aes,ecryptfs_key_bytes=16,ecryptfs_unlink_sigs)
/home/.shadow/a872706fda597de3825ae97a0ce39c3f6d557f3d/vault on /home/user/a872706fda597de3825ae97a0ce39c3f6d557f3d type ecryptfs (rw,nosuid,nodev,noexec,relatime,ecryptfs_sig=1a860241d16eec64,ecryptfs_fnek_sig=5316ba57b4965925,ecryptfs_cipher=aes,ecryptfs_key_bytes=16,ecryptfs_unlink_sigs)
/home/.shadow/a872706fda597de3825ae97a0ce39c3f6d557f3d/vault on /home/chronos/u-a872706fda597de3825ae97a0ce39c3f6d557f3d type ecryptfs (rw,nosuid,nodev,noexec,relatime,ecryptfs_sig=1a860241d16eec64,ecryptfs_fnek_sig=5316ba57b4965925,ecryptfs_cipher=aes,ecryptfs_key_bytes=16,ecryptfs_unlink_sigs)
/home/.shadow/a872706fda597de3825ae97a0ce39c3f6d557f3d/vault on /home/root/a872706fda597de3825ae97a0ce39c3f6d557f3d type ecryptfs (rw,nosuid,nodev,noexec,relatime,ecryptfs_sig=1a860241d16eec64,ecryptfs_fnek_sig=5316ba57b4965925,ecryptfs_cipher=aes,ecryptfs_key_bytes=16,ecryptfs_unlink_sigs)
/dev/sda1 on /var/run/crouton/usr/local/chroots/raring type ext4 (rw,relatime,commit=600,data=ordered)
tmpfs on /var/run/crw type tmpfs (rw,nosuid,nodev,noatime,size=128k,mode=700)
devtmpfs on /var/run/crouton/usr/local/chroots/raring/dev type devtmpfs (rw,nosuid,noexec,relatime,size=1992108k,nr_inodes=498027,mode=755)
devpts on /var/run/crouton/usr/local/chroots/raring/dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620)
shmfs on /var/run/crouton/usr/local/chroots/raring/dev/shm type tmpfs (rw,nosuid,nodev,noexec,relatime)
none on /var/run/crouton/usr/local/chroots/raring/sys type sysfs (rw,nosuid,nodev,noexec,relatime)
fusectl on /var/run/crouton/usr/local/chroots/raring/sys/fs/fuse/connections type fusectl (rw,nosuid,nodev,noexec,relatime)
tmp on /var/run/crouton/usr/local/chroots/raring/tmp type tmpfs (rw,nosuid,nodev,relatime)
none on /var/run/crouton/usr/local/chroots/raring/proc type proc (rw,nosuid,nodev,noexec,relatime)
tmpfs on /var/run/crouton/usr/local/chroots/raring/run type tmpfs (rw,nosuid,noexec,relatime,size=398648k,mode=755)
tmpfs on /var/run/crouton/usr/local/chroots/raring/run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k)
varrun on /var/run/crouton/usr/local/chroots/raring/var/host/dbus type tmpfs (rw,nosuid,nodev,noexec,relatime,mode=755)
varrun on /var/run/crouton/usr/local/chroots/raring/var/host/shill type tmpfs (rw,nosuid,nodev,noexec,relatime,mode=755)
varrun on /var/run/crouton/usr/local/chroots/raring/var/host/cras type tmpfs (rw,nosuid,nodev,noexec,relatime,mode=755)
/dev/mapper/encstateful on /var/run/crouton/usr/local/chroots/raring/var/host/timezone type ext4 (rw,nosuid,nodev,noexec,relatime,discard,commit=600,data=ordered)
/dev/root on /var/run/crouton/usr/local/chroots/raring/lib/modules/3.8.11 type ext2 (ro,relatime)
media on /var/run/crouton/usr/local/chroots/raring/var/host/media type tmpfs (rw,nosuid,nodev,noexec,relatime)
/home/.shadow/a872706fda597de3825ae97a0ce39c3f6d557f3d/vault on /var/run/crouton/usr/local/chroots/raring/home/wafflebox/Downloads type ecryptfs (rw,nosuid,nodev,relatime,ecryptfs_sig=1a860241d16eec64,ecryptfs_fnek_sig=5316ba57b4965925,ecryptfs_cipher=aes,ecryptfs_key_bytes=16,ecryptfs_unlink_sigs)
```

I appreciate any help you might be able to provide.
![playonlinux_001](https://f.cloud.github.com/assets/5723167/1727785/633257d6-62a4-11e3-8825-891d793ca83e.png)

**Comment by daradib** [comment]
I came across this same problem. Turns out PlayOnLinux would grep /etc/mtab for noexec by device instead of by filesystem, but firejail mounts the same device (/dev/dm-1) at different mount points, including /etc and /var with noexec option.

Was fixed by PlayOnLinux/POL-POM-4@6205938cd98092cf54418b2285bb2c241f341cdc.

**Comment by DennisLfromGA** [comment]
@daradib, Thanx for that news and link. 👍
